### PR TITLE
Convert ArgumentParser to a package in Chapel

### PR DESCRIPTION
**Pull request by arezaii**
This PR moves and renames the `MasonArgParse` library to `ArgumentParser` 
under `module/packages`. 

The scope of this work is described in cray/chapel-private#2481.

No parsing logic or other logic was updated, only the names/locations of files
and the order of use statements.

Use statements in Mason were updated to reflect the move. 

Tests were moved and updated in accordance with the new name and location of
the library.

TESTING:

Following this change and a `make cleanall`, the below checks are passing:

- [x] Can `make`
- [x] Can `make docs`
- [x] Can `make mason`
- [x] Can `make check`
- [x] Passing tests from `util/start_test test/mason/`
- [x] Passing tests from `util/start_test test/library/packages/ArgumentParser/`

Reviewed by @dlongnecke-cray 

Signed-off-by: Ahmad Rezaii ahmad.rezaii@hpe.com

### CODE REVIEW

**Comment by bradcray** [comment]
What is the impact of doing `make docs`?  Do we get docs for the new package module, or "not yet"?  (I can't recall whether it has to be added to an index file or Makefile to achieve that or not).

**Comment by arezaii** [comment]
> What is the impact of doing `make docs`? Do we get docs for the new package module, or "not yet"? (I can't recall whether it has to be added to an index file or Makefile to achieve that or not).

Not yet. It looks like it needs to be added to `modules/Makefile` in the `PACKAGES_TO_DOCUMENT` list. I am working on the docs, but I thought/hoped they could be added after feature freeze.

**Comment by bradcray** [comment]
> I thought/hoped they could be added after feature freeze.

They can, I was just curious / couldn't remember what the current behavior was.

**Comment by bradcray** [comment]
I gave this a quick look, and nothing jumped out at me as being problematic, but given the nature of the diff (lots of renamings, etc.) it wouldn't completely surprise me if I missed something either.  The testing you did seems appropriate, though if it were my PR, the paranoid part of me would probably do a full `paratest.chapcs` to make sure there were no surprises, esp. given that it's feature freeze day.

You can count this as a review or get someone else to give it a more careful one.

File: modules/packages/ArgumentParser.chpl [hunk 1]
```diff
@@ -17,11 +17,11 @@
  * limitations under the License.
  */
 
-module MasonArgParse {
-  private use List;
-  private use Map;
-  private use IO;
-  private use Sort;
+module ArgumentParser {
+  use List;
```
Comment (dlongnecke-cray) [inline modules/packages/ArgumentParser.chpl:21]: I can never remember, is the default access control for `use` private?

**Review by dlongnecke-cray — APPROVED** [review]
Looks good to me. Nice work!

File: modules/packages/ArgumentParser.chpl (hunk 1 above)
Comment (bradcray) [reply modules/packages/ArgumentParser.chpl:21]: It is now, yes.

Comment (dlongnecke-cray) [reply modules/packages/ArgumentParser.chpl:21]: Thanks!

**Comment by arezaii** [comment]
passed paratest

**Pull request merged**